### ISSUE TITLE
New Spinner Changes

### ISSUE THREAD

**Issue by GriffinOsu** [ppy/osu-stable-issues#1074]
The new spinner changes to stable have some glaring problems and inconsistencies attached with them, most notably, the fact that players are no longer able to pause on ninja spinners and must wait out the duration of them before the pause is inputted. 

Some examples of this happening: https://twitter.com/Mr_Falco/status/1600830907107389441?s=20&t=P-33PJr2KHTXkGvV-oBMKw
https://twitter.com/Emillyosu/status/1600822542016876545?s=20&t=P-33PJr2KHTXkGvV-oBMKw

There has also been another issue opened up about this which brings up this specific issue as well (https://github.com/ppy/osu-stable-issues/issues/1069), but this is only one of the issues. With the pause buffer added, we were told that you needed feedback on how this change affected spinners and the ability to replicate a certain pause bug (https://www.reddit.com/r/osugame/comments/zch8uf/comment/iyyts47/?utm_source=share&utm_medium=web2x&context=3), yet you had pushed the changes out to stable before getting a chance to get actual feedback from the only people who are affected by this issue in the first place (https://twitter.com/SevenEnd7/status/1600982008767074304?s=20&t=UlBAJqgDxnx_Nu82Z-1soA). Adding a pause buffer of this absurd length causes players who switch their grips or peripherals to get the most optimal spin (who have been doing this for years, as it has been the way the game has always been played) to no longer be able to achieve this and even become unable to pass the spinner with a 300 in cases of the spinner being extremely short. The pause buffer also not only hinders the way that people have been playing the game for years, but also does not fix the fact that spinners can still be outspun due to no changes being made to spinners themselves and this change has only had a negative effect on the community of people who have been pushing for change in the first place.

**Comment by peppy** [comment]
Tracking the "ninja spinner" issue at https://github.com/ppy/osu-stable-issues/issues/1069.

I understand your concern for the rest from a usability perspective, but understand that the only people I've heard complaints from are you and two other players. The overall reception is positive. We'll revisit balance of pausing on lazer, potentially disabling score submission if you pause outside of break time or similar. A discussion for another day.

**Comment by Exploity** [comment]
Understand that maybe you havent heard everyone who has complaints and the fact that the "positive reception" comes from people who don't understand the change and overall aren't even effected by it. The people this change is actually for all agree that this change is out right horrible.

If you don't want to out right revert the change at least make the buffer be at about the range of the original 32ms fix since a buffer of the same millisecond range as OD4 is WAY overkill.

(would have helped if you didn't push the current buffer straight to stable without any real feedback)

**Comment by pianodude1011** [comment]
It’s unfair to base your opinion of this having a “positive reception” when you stated that this change should not affect the majority of the playerbase — hence, they would have no reason to be upset about it. Also please bear in mind that the “only people complaining” is more than just two people, it’s just the ones who choose to say something about it, because many people I know are upset about it but few choose to talk about it/feel there is no way to voice their concerns, or that efforts will be futile.

Potentially unrelated, but did you ever see my comment mentioning that the buffer zone created is asymmetrical? Unsure if that was on purpose but if not fixing the center of the buffer would allow for a smaller buffer zone that is still relatively effective without being overly punishing.

**Comment by SevenEnd7** [comment]
Do you want spin players to tell you that the change wasn't good? I'll tell you right now that this change made me not want to play the game anymore. We have to change how we've been playing for YEARS over a single change that was proposed and greenlit OVERNIGHT. I even explained my concerns about this change before it was brought to stable. 

Why do we have to fight tooth and nail over issues that negatively impact the spinning community? You keep saying that you don't get enough feedback or very few people are complaining about the issues (in this case). I've tried showing you how these issues negatively impact our community, but you choose to "look at it later" or straight-up ignore it. Just because we're a small community doesn't mean you have to treat us this poorly. Not everyone wants to directly voice their opinions to a developer about why this change negatively impacts them. Especially when they see how he treats this community. 

I'm only writing this on GitHub because you're more likely to respond to issues here. Emailing you hasn't gotten any response (same for everyone else who did the same). Social media platforms end up causing more drama than necessary. And to be frank, I'm not sure what you expect us to do anymore. @peppy

**Comment by Gvppy** [comment]
I feel there should've been more testing before this went to stable. The current threshold for pausing on spinners is too strict. It currently makes some spinners unable to achieve max score on. I feel the threshold of -90 - +200 is too harsh, mainly the +200. The original of ~30 was fine but wasn't done the same as with the current change. If you made it -30 - +30 that would be better as it was your intended change.  But if it's easier I would say reverting to how spinners were before this change would be ideal for most players.

**Comment by KheopsOsu** [comment]
This change isn't good.

The spinning community is really disappointed about the change. Apparently "the only people I've heard complaints from are you and two other players." so I'm here to say that a lot more people than you think are really upset about this.

About not allowing breaks during the game, I personally would not like this change, not just for spins players, but also for people who need to fix their grip during the game, I am talking in particular mouse players to whom it can happen, and more.

Apologies for poor grammar. English is not my first language.

**Comment by ImMyyrh** [comment]
Honestly since the last spinner changes, I've found it very hard to play spinners anymore, and every other spin player I've shared thoughts with about this current situation, we get to the exact same point, which is a very hasty change. I really doubt its has a "positive overall reception", as it affects only a very small playerbase and we all share the same thoughts.

This current changes, makes some spinners *unable* to achieve max score on, and outspins CAN still happen without triggering it. So this barely changed anything, only made it harder for us spin players to play the way we used to do (and its killing the motivation for every spin player to go for such scores, including myself).

Although I appreciate your effort into this situation, I would suggest to revert the change, as it realistically changes nothing to the main topic discussion, which are "outspins". Or at least revert it to the original 32ms fix proposed at the start, until theres another fix or anything proposed.

**Comment by Emillyosu** [comment]
As someone who has a super unstable grip, I often find myself pausing on spinners/breaks to regrip my pen. This 200ms delay is awful. I can easily feel the delay. It kinda feels as if my game is stuttering/lagging whenever I try to pause

**Comment by Shaper021** [comment]
I made a github account only so I could comment a piece of my mind on this subject too, because this change affects how i play the game on a daily basis. I think theres no need for me to mention a lot of stuff here like the issue with pausing on really short spinners (ex. ninja spinners) because everythings already been said, but I will say that this makeshift "spin fix" is just useless and is honestly only aimed to affect the minority of players (aka less than 100 players/aka us) and it still hasnt fixed anything. The way we play the game has pretty much been ruined by this change and it demotivated a lot of players from playing the same game we used to grind for hours. It pretty much made it almost impossible to pause on time whenever theres a spinner coming up in the map which makes harder spins absolutely impossible to cap (max out), which is kinda what, we as a community, like to do. I understand that we are a minority of the playerbase but we decided to speak up because it only affects us and rarely anyone else and we dont want this pause delay in the game, hoping for a revert to this "fix".

**Comment by LapplandOsu** [comment]
I just made a github acc just for this. This change is killing the meta of how spinning community play in general. This delay is not the solution for the "outspin" issue, because outspinning without pausing is still possible. I hope this change gets reverted or "fixed" as fast as possible.

**Comment by okapp1** [comment]
Thank you very much for making this game interesting through the constant and unflagging development by you and the rest of the development team. However, the recent sudden change in the specifications regarding the pause just before the spinner has made the game less enjoyable, not only for me and those who have expressed their opinions on this page, but for many others who are sincere about the spin. I believe that this change should be withdrawn.

　The main reason why the game has become more boring is because of the recent updates to the spinner, which have made the game much less manageable than it used to be, since the action to pause just before the spinner, which used to pause at the same time as the action, now does not pause immediately. This is because the operation is now no longer paused immediately, and is much less easy to handle than before. As many have already mentioned in various places, we have spent a long period of time researching and improving our own grip for the spinner so that we can get the most possible score we can get from the spinner, and then spinning the spinner with that grip. To achieve this, we have to pause before the spinner and change from our normal grip to a grip that is appropriate for the spinner. However, recent updates to the spinner have made it impossible for us to pause comfortably. This means that it is no longer possible to get a satisfactory score on a certain spinner of a map.

　Furthermore, the update should have fixed the spinner-related bug. However, it nevertheless had a significant negative impact on players who are simply spinning the spinner with great effort. This is an undesirable update for us, and it also tramples on the efforts of these players.

　As such, I would like to request that this change be rescinded. Of course, there may be developmental reasons for this outcome, but it is very troubling for our situation.

**Comment by wilson0512** [comment]
I don't like this change.

The pause buffer is too long. As a spin player, I had to pause the game in order to change my grip for spinning (not only me need to do this). But now, I can't pause before the spinner starts because of the pasue delay. Therefore, it is unable to reach the max spinner score anymore. It just killed the playstyle for most of the spin players.

This changes just killing the motivation of people on playing. I hope the change can be revert or shorten the pause delay to acceptable level.

**Comment by Kotoki1337** [comment]
i don't really think using this method to blocking players getting outspin without fixing it itself is the right thing to do. there are still new outspin scores after this change, and the change brings more and more new problems into the game. most players who give positive feedback don't even finish spinning the spin seriously, i see no one who is really focused on spinning (most people never trigger an outspin) has a positive attitude toward this. you guys really need to look into whether this change is correct.

**Comment by itzEddd** [comment]
Hello peppy. I am writing this in the hope that there is someone on the other end of the screen that can at least try to look at this situation from our end.  The new pause delay added to prevent people from "outspinning auto" was extremely uncalled for and did nothing to actually fix the "spin rate bug." Instead it caused unrelated issues for spin players that require the pause for changing grip and adjusting their tablet. To you it may seem like we are niche community and because it's so few of us, it shouldn't really matter right? Well that's why me and the others in this comment section are expressing our opinions. We don't think that this change is fair as it's been how many of us have played the game for years. I understand why you think it's the simplest course of action, but an overnight bandage fix is definitely not how this problem should be tackled.  If it's just not possible to completely fix the main issue brought up by sevenend7 without completely overhauling the game's code, at least look into lowering the pause delay to an acceptable amount (until a better fix is implemented.) As it is, the pause delay is set to an extremely high amount and kills off a whole subset of the community.  

I can tell you have a lot on your plate with lazer development, but I heavily implore you to look into how to actually solve the problem instead of punishing some of the most dedicated people that play your game.

**Comment by Brozeus** [comment]
I tried the new spin changes and didn't like it, this change made it impossible to get the max score on many maps since the delay is too much, please consider reverting this change since it affects every spin player

**Comment by MiguelShadow** [comment]
Pretty much everything has been said already so im just giving my personal thoughts.

Spinning is super tiring, but if ive to pause spin to match others people score i will because thats my playstyle and prob others too.
Ever since the change was made I slighly stopped playing the way I do as it became way more frustrating than it is for spinning, like even for a player like me who cant max spin or accidentally outspin but still needs to pause to match other scores and get this much delay its just not fun anymore.

**Comment by Fleurs1** [comment]
I make a Github account and come here for letting you know what you said "the overall reception is positive" is controversial.

Many spin players like me don't like this update, we feel the same way. The strict pause limit makes it impossible to get max score in some spinners anymore, which also destroys the game experience of spin players. Imagine that fps game players have to encounter a 200ms delay when aiming then fire (moving and pause in osu). This change of course reduces the motivation of players who grind at spinners and leaderboards. In the past, I could repeatedly play dozens and hundreds of times in order to make one more turn on the spinners. Now, as long as there is a delay in one of the spinners in the maps, players who are pursuing full marks will ofc fail in that try, that's very depressing.

I highly hoping this change can be revert as soon as possible or making better changes later.

**Comment by lan641** [comment]
As someone who pause spins, this change is just awful. It just makes pause spinning worse and doesn't even fix outspinning infact it's still possible to outspin even with this pause delay.

Also disabling score submission for pausing outside of breaks is a bad idea. Imagine having your top pp play or good score run not sumbitting just because you paused outside of a break to either change/fix grip or to do something else. It's just to punishing and most players would dislike this feature if that were to be implemented.

I really hope this delay gets reverted even if pause spinning is considered a niche community by most of the player base.

**Comment by xYohane** [comment]
Could you please revert it back to <35ms. This doesn't fix the outspinning issue but it will probably for pause spin and ninja spinner bug.

**Comment by loveGirlsOSU** [comment]
I don't like this spin change. Achieve max score on many spins not possible anymore, also less comfortable and does not fixing the issue with outspinning. In conclusion, update has only negative affects.

**Comment by KeplerSync** [comment]
I don't like this change, this isnt fixing anything, its a way of desmotivate a huge part of the leaderboard community, making hard working and dedication, useless. Almost any spin player can tell the effort to develop a great spin grip, to get way into this world of spin. The spin players and leaderboard community is really affected with this change, making a lot of us, giving up. 
I hope this delay gets revert.

**Comment by Soroic416** [comment]
This spinner change made me play less.

I always change my grip right before the spinners as other players do but this 200ms delay is too much. In some maps, I can't even get close to cap scores anymore. The change demotivated me so i don't play much for leaderboards like i used to. I don't know how to prevent players from outspinning but this isn't the right solution.

Please revert the spinner change.

**Comment by aalleetthheeaa** [comment]
i'v been losing motivation to play recently cause of spinner "change" , adding delay on pausing during spinner didnt fix anything except make it more way harder and uncomfortable to play. just revert the spinner back and i would really appreciate it cheerrss ...

**Comment by 156752571** [comment]
There was new spinner update which made spinner delay a bit. And because of it, we get fewer points from spinners when note right before spinner is at the corner (sorry for bad english, korean is my first language)."

**Comment by psychokeu** [comment]
This pause delay modification has not effectively helped, rather it has affected not only how I play, but many other spin players alike in the community. As explained to you for what feels like the billionth time, pause spinning is not meant for any intent to outspin auto, rather to fix your grip or change your peripherals. However, it feels like that information went in one ear and out the other, because what you decided to do was to send an rushed update overnight that didn’t even get tested before getting on to stable, which set an immense ms delay on when you can pause on a spinner. These delay values came out extremely absurd and has essentially killed the ability to get max score on a mass selection of maps. And of course I know you’re a busy man, but please take some time to listen to us as a community and make these values more tolerable.

In short, I and many others do not like this update and how it was both made and released.

**Comment by Marlon-2304** [comment]
I dont like the new change and it doesnt really help.
200 ms delay is way too long. You can't even reach the maximum score because of it. Also, some spinners can still be paused normally like before, which doesn't make sense. The only thing this change does is killing our playstyle and our motivation.

I hope the delay gets reverted.

**Comment by HeyCatia** [comment]
I am certainly not a fan of the change made to combat players abusing spinner bugs. 200ms is an excessive amount of buffer time to place on pausing. To put it into perspective, 200ms is the entire hit window for an object at OD10, that is a large and noticeable period of time. As suggested in comments above, it would be great if the buffer time could be removed entirely or reduced to a much more reasonable amount instead.

As for the statement that “the overall reception is postitive.”. This change was made as a hotfix targeted for a specific portion of players, in turn the players who were not the main target of this change had no issues with it. On the other hand, the player base affected by this change has had an overwhelmingly negative response towards it. All of the leaderboard players I have talked to thus far have taken issue with this change.

 I kindly hope this clears up the notion that only 2 or 3 players have complaints regarding the pause buffer time.

**Comment by creicerbros** [comment]
This update does not solve the outspin and complicates the whole spinnig process because of the long pause delay, not really get the max score on most spinners. xd

**Comment by alyhisa** [comment]
I don't like the new change, It didn't change anything it just made impossible to cap some spinners, 200ms delay is too much. And I think it's unfair. 

I hope the delay gets reverted.

**Comment by Mr-Falco** [comment]
In my opinion, this update is rushed and should've been refined / discussed more with the players affected by it. ~200ms is an unnecessarily large amount of delay, which makes the game feels choppy/broken when pausing before a spinner.

A smaller or more optimized amount of delay, to a point where the outspinning is prevented and the delay is bearable can be figured out.

**Comment by Piixers** [comment]
Hi peppy, I pretty much agree with everything other people said as this change is too drastic and it makes some spinners impossible to cap, which is not fun. I think a revert would be the best option or at least decrease the delay when we pause; it's currently unsustainable on some maps.

**Comment by RimuwuOsu** [comment]
This "fix" should be reverted because it doesn't prevent people from outspinning auto at all. I can still spin with no pause but some spinners are just impossible to cap without pausing and after this change there is no way to cap them.

**Comment by GriffinOsu** [comment]
Thank you for the response Peppy. I understand that while it may seem like the community as a whole is viewing these changes in a positive light, it is due to the fact that most of the community is not affected by this issue. The community of leaderboard farmers and spin players whose playstyle is directly affected by this change on the other hand, seem to mostly view these recent changes in a negative way. I also understand your concern for not reverting these changes, but as seen in the many comments above, the dedicated community of players who are actually affected are speaking out and giving you the feedback that you had asked for about these changes.

We are a relatively small community, but that doesn't mean that we should be punished for playing the game differently than most other players.

**Comment by Auxitly** [comment]
Honestly, regarding about the current 'spinner update', this fix doesn't solve anything at all, in fact, it just made pause spinning a LOT worse. I'm just hoping that you would revert this change to what it was before all of this.

Pretty much the rest of the people here are also saying the same thing, it actually made capping spinners from somewhat possible to downright impossible. What this also does is that it doesn't fix the whole "outspin" problem at all, you can still do that without pausing.

I do think that this update is a change in the right direction, but the way you just made the change is just so rushed and felt like it needed more time and testing before you put it in Stable and the game as a whole. And that being said, the +200ms delay in pausing is insane the way you look at it, it literally just made maps that has ninja spinners and maps that were considered to be doable to cap the spinners on is no longer possible because of this change. 

Please revert this change or think about lowering the delay so that it's still somewhat tolerable to the point you can still cap spinners with the use of pause spinning. Thanks.

**Comment by plejserxd** [comment]
This spinner "fix" would be probably alright if the delay wasnt that big. However i think that its still not best way to fix it this way because its still possible to outspin it really dosnt solve the issue. Currently this change only makes spinning annoying and some spinners are even impossible to cap now days.

Please revert this change or find a better way to solve the issue.

**Comment by vYper2011** [comment]
This new spinner change made me feel much less motivated to grind leaderboard scores on easier diffs. Adding a 200ms pause delay right before a spinner does not fix the issue with outspinning (most notably, getting above max score). It instead makes the game feel much more uncomfortable to play. This change also made it impossible to get the max score on many beatmaps. I suggest that this new spinner update gets reverted to what it was like before all of this happened.
